### PR TITLE
feat: add support for friendliai dedicated endpoint

### PR DESCRIPTION
**Pull request by pocca2048**
## Title
[FriendliAI Dedicated Endpoint](https://friendli.ai/products/dedicated-endpoints/) support

## Type
🆕 New Feature
📖 Documentation

## Changes
- add support for friendliai dedicated endpoint (previously, it only supported serverless endpoint)

## Description
- Serverless Endpoint : Users can access pre-configured models seamlessly, without any need for setup or configuration. This endpoint allows for immediate utilization of LLMs, providing a streamlined and efficient user experience.
- Dedicated Endpoint : Users have the capability to deploy and manage their own custom models by creating dedicated endpoints. This option requires specifying a different API base and offers greater flexibility and control over model deployment and management.

## [REQUIRED] Testing - Attach a screenshot of any new tests passing local
If UI changes, send a screenshot/GIF of working UI fixes
```python
from litellm import completion
import os

ENDPOINT_ID = ""
os.environ['FRIENDLI_API_BASE'] = "https://inference.friendli.ai/dedicated/v1/"
os.environ['FRIENDLI_TOKEN'] = ""
response = completion(
    model=f"friendliai/{ENDPOINT_ID}", 
    messages=[
       {"role": "user", "content": "hello from litellm"}
   ],
)
print(response)
```

image

### CODE REVIEW

File: litellm/utils.py [hunk 1]
```diff
@@ -4020,7 +4020,10 @@ def get_llm_provider(
                     or get_secret("TOGETHER_AI_TOKEN")
                 )
             elif custom_llm_provider == "friendliai":
-                api_base = "https://inference.friendli.ai/v1"
+                api_base = (
+                    get_secret("FRIENDLI_API_BASE")
```
Comment (ghost) [inline litellm/utils.py]: hey @pocca2048 this is missing support for a user dynamically passing in the api base - 

https://github.com/BerriAI/litellm/blob/72045cc0c5f27e170725f57636c87ac739dbc22b/litellm/utils.py#L3994

**Review by ghost — CHANGES_REQUESTED** [review]
please add support for dynamically passing in api_base and api_key - 

https://github.com/BerriAI/litellm/blob/72045cc0c5f27e170725f57636c87ac739dbc22b/litellm/utils.py#L3994

File: litellm/utils.py (hunk 1 above)
Comment (pocca2048) [reply litellm/utils.py]: Thank you for the suggestion! I have made changes.

**Comment by pocca2048** [comment]
@krrishdholakia Hi, could you review this PR again for me? Thanks!

**Pull request merged**